handlers.go

```diff
@@ -33,6 +35,39 @@ func (h *DeviceHandlers) devicesHandler(w http.ResponseWriter, r *http.Request)
 		return
 	}
 
+	dh.sendDeviceObject(w, device)
+
+	return
+}
+
+func (dh *DeviceHandlers) getDeviceHandler(w http.ResponseWriter, r *http.Request) {
+	input := mux.Vars(r)["id"]
+	id, err := strconv.Atoi(input)
+	if err != nil {
+		w.WriteHeader(http.StatusBadRequest)
+		fmt.Println(err.Error())
+		return
+	}
+
+	device, err := dh.service.GetDevice(id)
+	if err != nil {
+		switch err.(type) {
+		case ErrNotFound:
+			w.WriteHeader(http.StatusNotFound)
+			fmt.Println(err.Error())
+		default:
+			fmt.Println(err.Error())
+			w.WriteHeader(http.StatusInternalServerError)
+		}
+		return
+	}
+
+	dh.sendDeviceObject(w, device)
+
+	return
+}
+
+func (dh *DeviceHandlers) sendDeviceObject(w http.ResponseWriter, device *Device) {
```

Comment:
I have a rename suggestion: `sendDeviceObject` to `writeObject` or `respondWithObject`.

`send` suggests that you are connecting somewhere and pushing the object but in fact you are just responding on connection initiated from the client side.